apps/estuary/test/integration/estuary_test.exs

```diff
@@ -0,0 +1,134 @@
+defmodule Estuary.EstuaryTest do
+  use ExUnit.Case
+  use Placebo
+  use Divo
+  import SmartCity.TestHelper, only: [eventually: 1]
+
+  alias Estuary.EventTableHelper
+
+  @elsa_endpoint Application.get_env(:estuary, :elsa_endpoint)
+  @event_stream_topic Application.get_env(:estuary, :event_stream_topic)
+  @event_stream_table_name Application.get_env(:estuary, :event_stream_table_name)
+
+  test "should create topic when estuary starts" do
+    assert Elsa.Topic.exists?(@elsa_endpoint, @event_stream_topic)
+  end
+
+  test "should create event_stream table and confirm all the column exists when estuary starts" do
+    expected_columns = [
+      ["author", "varchar", "", ""],
+      ["create_ts", "bigint", "", ""],
+      ["data", "varchar", "", ""],
+      ["type", "varchar", "", ""]
+    ]
+
+    actual_columns =
+      "DESCRIBE #{@event_stream_table_name}"
+      |> Prestige.execute()
+      |> Prestige.prefetch()
+
+    assert expected_columns == actual_columns
+  end
+
+  test "should persist event to the event_stream table" do
+    produce_event(
+      @event_stream_topic,
+      %{
+        "author" => "reaper",
+        "create_ts" => 5,
+        "data" => "some data for reaper",
+        "type" => "some type for reaper"
+      }
+      |> event_struct()
+    )
+
+    expected_value = [["reaper", 5, "some data for reaper", "some type for reaper"]]
+
+    eventually(fn ->
+      actual_value =
+        "'reaper'"
+        |> EventTableHelper.select_table_data()
+
+      assert expected_value == actual_value
+    end)
+
+    EventTableHelper.delete_table_data()
+  end
+
+  test "should persist batch of events to the event stream" do
+    produce_event(@event_stream_topic, [
+      %{
+        "author" => "forklift",
+        "create_ts" => 1,
+        "data" => "some data for forklift",
+        "type" => "some type for forklift"
+      }
+      |> event_struct(),
+      %{
+        "author" => "valkyrie",
+        "create_ts" => 2,
+        "data" => "some data for valkyrie",
+        "type" => "some type for valkyrie"
+      }
+      |> event_struct()
+    ])
+
+    expected_value = [
+      ["forklift", 1, "some data for forklift", "some type for forklift"],
+      ["valkyrie", 2, "some data for valkyrie", "some type for valkyrie"]
+    ]
+
+    eventually(fn ->
+      actual_value =
+        "'forklift', 'valkyrie'"
+        |> EventTableHelper.select_table_data()
+
+      assert expected_value == actual_value
+    end)
+
+    EventTableHelper.delete_table_data()
```

Comment:
This probably belongs in an [ExUnit `on_exit` Callback](https://hexdocs.pm/ex_unit/ExUnit.Callbacks.html) since this won't run if this test fails which could cause invalid cascading failures.

Reply:
Agreed